packages/core/src/common/utils/reactUtils.ts

```diff
@@ -92,12 +92,12 @@ export function isReactElement<T = any>(child: React.ReactNode): child is React.
     );
 }
 
-function isReactFragment(child: React.ReactNode): child is React.ReactFragment {
+function isReactFragment(child: React.ReactNode): child is Iterable<React.ReactNode> {
```

Comment:
For others, it looks like replacing `ReactFragment` with `Iterable<React.ReactNode` is what a lot of React 18 upgrade codemods do as well:

https://github.com/eps1lon/types-react-codemod/blob/main/CHANGELOG.md#minor-changes-4